py4cast/datasets/smeagol.py

```diff
@@ -779,3 +793,130 @@ def domain_info(self) -> DomainInfo:
+                        sample
+                        * self.settings.num_total_steps : sample
+                        * self.settings.num_total_steps
+                        + self.settings.num_input_steps
+                    ]
+
+                    output_terms = [
+                        input_terms[-1] + self.settings.term["timestep"] * (step + 1)
+                        for step in range(self.settings.num_inference_pred_steps)
+                    ]
+
+                    samp = InferSample(
+                        date=date,
+                        member=member,
+                        input_terms=input_terms,
+                        output_terms=output_terms,
+                    )
+
+                    if samp.is_valid(self.params):
+
+                        samples.append(samp)
+                        number += 1
+        print("All samples are now defined")
+
+        return samples
+
+    @classmethod
+    def from_json(
+        cls,
+        fname: Path,
+        num_input_steps: int,
+        num_pred_steps_train: int,
+        num_pred_steps_val_tests: int,
+        config_override: Union[Dict, None] = None,
+    ) -> Tuple[None, None, "InferSmeagolDataset"]:
+        """
+        Return 1 InferPoesyDataset.
+        Override configuration file if needed.
+        """
+        with open(fname, "r") as fp:
+            conf = json.load(fp)
+            if config_override is not None:
+                conf = merge_dicts(conf, config_override)
+
+        grid = Grid(**conf["grid"])
+        param_list = []
+        # Reflechir a comment le faire fonctionner avec plusieurs sources.
+        for data_source in conf["dataset"]:
+            data = conf["dataset"][data_source]
+            members = conf["dataset"][data_source].get("members", [0])
+            term = conf["dataset"][data_source]["term"]
+            param_grid = Grid(**data["grid"])
+            for var in data["var"]:
+                vard = data["var"][var]
+                # Change grid definition
+                if "level" in vard:
+                    level_type = "hPa"
+                    var_file = var
+                else:
+                    level_type = "m"
+                    var_file = "surface"
+                param = Param(
+                    name=var,
+                    levels=vard.pop("level", [0]),
+                    grid=param_grid,
+                    level_type=level_type,
+                    fnamer=partial(
+                        smeagol_forecast_namer,
+                        model=data["grid"]["model"],
+                        domain=data["grid"]["domain"],
+                        geometry=data["grid"]["geometry"],
+                        var=var_file,
+                    ),
+                    **vard,
+                )
+                param_list.append(param)
+
+        inference_period = Period(**conf["periods"]["test"], name="infer")
+        ds = InferSmeagolDataset(
```

Comment:
You put the dataset instantiation in the for loop for Poesy, shouldn't be the same here ?

Reply:
Mhh I don't think so ? The for loop is used to add all parameters to a list. The dataset's instantiation takes this list as an input. For me Poesy and Smeagol are both instantiate dataset outside the loop.

Reply:
Ok for me, but I mean there is a difference between the for loop in `poesy.py` (instantiation is in the _for_ loop) and `smeagol.py` file (here instantiation is after the _for_ loop). So please fix the mis-indentation in `poesy.py`

Reply:
Oh okay! thanks for noticing it. It's fixed.